app/src/main/java/io/unityfoundation/dds/permissions/manager/security/AuthConfigService.java

```diff
@@ -0,0 +1,114 @@
+// Copyright 2023 DDS Permissions Manager Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package io.unityfoundation.dds.permissions.manager.security;
+
+import io.micronaut.context.annotation.Property;
+import io.micronaut.context.env.Environment;
+import io.micronaut.http.HttpResponse;
+import jakarta.inject.Singleton;
+import org.bouncycastle.asn1.pkcs.PrivateKeyInfo;
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+import org.bouncycastle.openssl.jcajce.JcaPEMWriter;
+import org.bouncycastle.util.io.pem.PemObject;
+import org.bouncycastle.util.io.pem.PemReader;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.security.KeyFactory;
+import java.security.PublicKey;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.X509EncodedKeySpec;
+
+@Singleton
+public class AuthConfigService {
+
+    @Property(name = "permissions-manager.application.jwt.signature.public")
+    protected String publicKey;
+
+    @Property(name = "permissions-manager.application.jwt.signature.private")
+    protected String privateKey;
+
+    private final Environment environment;
+
+    public AuthConfigService(Environment environment) {
+        this.environment = environment;
+    }
+
+    public HttpResponse<?> getPublicKeyText() throws Exception {
+        String publicKey = objectToPEMString(readX509PublicKey());
+        return HttpResponse.ok(publicKey);
+    }
+
+    public RSAPublicKey readX509PublicKey() throws Exception {
+        KeyFactory factory = KeyFactory.getInstance("RSA");
+
+        File file;
+        if (environment.getActiveNames().contains("dev") || environment.getActiveNames().contains("test")) {
+            file = getFileFromResource(publicKey);
+        } else {
+            file = new File(publicKey);
+        }
```

Comment:
If I'm understanding correctly, you are suggesting that we load from resource as a fallback?

The intent of the code is to use the keys `app/src/main/resources` directory only in dev or test environments (references to keys in resource directory are made in application-dev and application-test yml files). Otherwise, the app fails to start as the environment variables must be declared.

Reply:
I think I typed up a response to this and lost it.

Yes.  The code should attempt to treat the paths as files first, which is what most users will do, then fallback to treating the paths as resources which will help for dev and test.  In this way, the environment specific logic goes away.